### PR TITLE
Implemented cypress integration testing. 

### PR DESCRIPTION
**Pull request by Txu045**
Example cypress tests are left in the codebase for now for reference, will remove later. 

New cypress test suit in yosoko > index.spec.js.

Run `npm run dev` for localhost to run in the background
Run `npm run cypress` and manually click the popup with yosoko > index.spec.js

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,6 +54,11 @@ npm run test:unit:components
 
 # run Firestore security rules unit tests
 npm run test:unit:firestore
+
+# TODO make cypress tests run automatically, currently manual local tests with plenty of examples
+# Delete all examples once we are comfortable writing cypress tests
+# run Cypress integration tests - will open new window. Manually select yosoko > index.spec.js test suit
```
[the remark this answers is not in the document]
Comment (Txu045) [reply README.md]: added in npm run cypress:all which runs all tests in headless mode. Next commit will include a workflow yml file hopefully for automatic testing.

File: README.md [hunk 2]
```diff
@@ -54,6 +54,13 @@ npm run test:unit:components
 
 # run Firestore security rules unit tests
 npm run test:unit:firestore
+
+# TODO make cypress tests run automatically, currently manual local tests with plenty of examples
```
Comment (sttic) [inline README.md:58]: Nitpick: todo message not needed anymore

File: .github/workflows/integration-test-components.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      # Install NPM dependencies, cache them correctly
+      # and run all Cypress tests
+      - name: Cypress run
+        uses: cypress-io/github-action@v2
+        with:
+          build: npm run build
+          start: npm start
```
Comment (sttic) [inline .github/workflows/integration-test-components.yml:17]: I looked into the [failed run](https://github.com/yokoso-capstone/yokoso/runs/4086370827?check_suite_focus=true) by running the action locally (using this [cool tool](https://github.com/nektos/act)) and turned out to be an issue with mismatched npm versions. So the fix is to keep everything consistent with the package-lock.json version and build env by adding the following steps at the beginning, like how the other workflows include: 

```yml
      - uses: actions/checkout@v2
      - name: Setup node
        uses: actions/setup-node@v2
        with:
          node-version: "12"
      - name: Install npm v7
        run: npm install -g npm@7
```

So the whole thing would look like:

```yml
name: Cypress Tests
on: push

jobs:
  cypress-run:
    name: Test components with Cypress
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - name: Setup node
        uses: actions/setup-node@v2
        with:
          node-version: "12"
      - name: Install npm v7
        run: npm install -g npm@7
      # Install NPM dependencies, cache them correctly
      # and run all Cypress tests
      - name: Cypress run
        uses: cypress-io/github-action@v2
        with:
          build: npm run build
          start: npm start
```

Note: I ran it locally using a minimal Docker container where I had to manually install some [Cypress dependencies](https://docs.cypress.io/guides/continuous-integration/introduction#Dependencies). The workflow might have to explicitly install those if the GitHub Actions or `cypress-io/github-action` workflow environment doesn't include them.

**Review by sttic — COMMENTED** [review]
Thanks for the updates Tommy!

I left some comments -- hopefully they're quick to look at :sweat_smile: 

Also consider adding `cypress/videos/` to .gitignore so we won't accidentally commit the video files Cypress makes after running the tests.

File: cypress/integration/yosoko/index.spec.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  context("with Vancouver searched", () => {
+    beforeEach(() => {
+      cy.visit(
+        "http://localhost:3000/search?center=-123.113953&center=49.260872&place=Vancouver%2C+British+Columbia%2C+Canada&text=Vancouver"
+      );
+    });
+
+    it("selects sample listing", () => {
+      cy.contains("2bdrm", { timeout: 4000 }).click();
+      cy.contains("Lease details", { timeout: 2000 });
```
Comment (sttic) [inline cypress/integration/yosoko/index.spec.js:95]: I had some issues running locally while emulating GitHub Actions where it failed on this line, maybe due to the new page not loading in time.

Consider adding/using the following which helped and also tests the route change.

```javascript
      cy.location("pathname", { timeout: 4000 }).should("include", "/listings");
      cy.contains("Lease details");
```

**Pull request merged**